spec/java/util/stream/DoubleStream.main.lsl

```diff
@@ -0,0 +1,1266 @@
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        result = _findFirst();
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.iterator (@target self: DoubleStream): PrimitiveIterator_OfDouble
+    {
+        result = new DoubleStreamIteratorAutomaton(state = Initialized,
+            parent = self,
+            cursor = 0,
+        );
+    }
+
+
+    fun *.spliterator (@target self: DoubleStream): Spliterator_OfDouble
+    {
+        action TODO();
+    }
+
+
+    // within java.util.stream.BaseStream
+    fun *.isParallel (@target self: DoubleStream): boolean
+    {
+        result = this.isParallel;
+    }
+
+
+    fun *.sequential (@target self: DoubleStream): DoubleStream
+    {
+        this.isParallel = false;
+        result = self;
+    }
+
+
+    fun *.parallel (@target self: DoubleStream): DoubleStream
+    {
+        this.isParallel = true;
+        result = self;
+    }
+
+
+    fun *.unordered (@target self: DoubleStream): DoubleStream
+    {
+        result = self;
+    }
+
+
+    // within java.util.stream.BaseStream
+    fun *.onClose (@target self: DoubleStream, arg0: Runnable): DoubleStream
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        val listLength: int = action LIST_SIZE(this.closeHandlers);
+        action LIST_INSERT_AT(this.closeHandlers, listLength, arg0);
+        result = self;
+    }
+
+
+    // within java.lang.AutoCloseable
+    fun *.close (@target self: DoubleStream): void
+    {
+        val listLength: int = action LIST_SIZE(this.closeHandlers);
+
+        // UtBot note: this implementation does not care about suppressing and throwing exceptions produced by handlers
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, listLength, +1,
+            _closeHandlers_loop(i)
+        );
+
+        this.closeHandlers = action LIST_NEW();
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    @Phantom proc _closeHandlers_loop (i: int): void
+    {
+        val currentHandler: Runnable = action LIST_GET(this.closeHandlers, i) as Runnable;
+        action CALL(currentHandler, []);
+    }
+
+
+    fun *.dropWhile (@target self: DoubleStream, predicate: DoublePredicate): DoubleStream
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (predicate == null)
+            _throwNPE();
+
+        if (this.length == 0)
+        {
+            val emptyStorage: array<double> = action ARRAY_NEW("double", 0);
+            result = new DoubleStreamAutomaton(state = Initialized,
+                storage = emptyStorage,
+                length = 0,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+        else
+        {
+            var dropLength: int = 0;
+
+            var i: int = 0;
+            action LOOP_FOR(
+                i, 0, this.length, +1,
+                _dropWhile_loop(i, dropLength, predicate)
+            );
+
+            val newLength: int = this.length - dropLength;
```

Comment:
What if `newLength` is equal to `0` (zero)? I see an opportunity to avoid a following loop down below.

Reply:
Fixed;

Reply:
Incorrect type of automaton is being returned: `IntStreamAutomaton` instead of `DoubleStreamAutomaton`.

Reply:
Fixed